### PR TITLE
It testing

### PR DESCRIPTION
**Pull request by chrissharp80**

### CODE REVIEW

File: integration_test/integration_functions_lib.go [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	LogPrefix string
+	Logger    *log.Logger
```
Comment (mrshah-at-ibm) [inline integration_test/integration_functions_lib.go:22]: Can we add a default logger if a logger is not set?

File: integration_test/integration_functions_lib.go [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+	}
+	Logger.Println(LogPrefix + "**SUCCESS** - CA created")
+
+	// as a last step, we'll wait on the CA to come up before allowing anything else to happen
+	err = waitForCaToComeUp(*result.ApiURL)
+	return *result.Msp.Component.TlsCert, *result.ApiURL, err
+}
+
+func GetDecodedTlsCert(ec string) ([]byte, error) {
+	// decode the base64 string in the CA's MSP
```
Comment (mrshah-at-ibm) [inline integration_test/integration_functions_lib.go:69]: The comment about what the function does should be put above the function.
Ref: https://golang.org/src/go/doc/example.go

File: integration_test/integration_functions_lib.go [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+
+	start := time.Now()
+
+	client := http.Client{Timeout: 5 * time.Second} // create a client to handle the Get requests - this allows us the timeout
+
+	deadline := start.Add(10 * 60 * time.Second) // ten minutes
+
+	withinDeadline := false // final check at the end
+
+	for !time.Now().After(deadline) { // make sure we're not past our five minute deadline
```
Comment (mrshah-at-ibm) [inline integration_test/integration_functions_lib.go:282]: looks like we are setting up a 10 minutes deadline

File: integration_test/integration_functions_lib.go [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+		Logger.Println(LogPrefix + "CA's cainfo polled ")
+		resp, err := client.Get(apiUrl + "/cainfo")
```
Comment (mrshah-at-ibm) [inline integration_test/integration_functions_lib.go:284]: is this loop going to run with no sleep? we should look at using Poll or ticker.

File: integration_test/integration_test.go [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+	it.LogPrefix = "[IT_TEST] "
+	it.Logger.Println(ItTest + "\n\n***********************************STARTING INTEGRATION TEST***********************************")
+
+	// get global setup information from a file
+	setupInfo = setupInformation{}
+	err := getSetupInfo(&setupInfo)
+	Expect(err).NotTo(HaveOccurred())
+
+	// create a blockchain service to work with
+	service, err = createAService(setupInfo)
```
Comment (mrshah-at-ibm) [inline integration_test/integration_test.go:76]: we are creating a blockchain "service instance". We can update the comment.

**Review by mrshah-at-ibm — CHANGES_REQUESTED** [review]

File: integration_test/integration_functions_lib.go (hunk 1 above)
Comment (chrissharp80) [reply integration_test/integration_functions_lib.go:22]: done

File: integration_test/integration_functions_lib.go (hunk 2 above)
Comment (chrissharp80) [reply integration_test/integration_functions_lib.go:69]: This comment is about that function call to decode the string. It's mainly there to quickly see where the base64 string came from. The function returns it in decoded form. I see your point though, given the name of the function. Moving the comment now.

File: integration_test/integration_functions_lib.go (hunk 3 above)
Comment (chrissharp80) [reply integration_test/integration_functions_lib.go:282]: fixed

File: integration_test/integration_functions_lib.go (hunk 4 above)
Comment (chrissharp80) [reply integration_test/integration_functions_lib.go:284]: I put it to sleep for 2 seconds each time through use `time.sleep()`.

File: integration_test/integration_test.go (hunk 5 above)
Comment (chrissharp80) [reply integration_test/integration_test.go:76]: done

**Review by dshuffma-ibm — CHANGES_REQUESTED** [review]
I'd like this integrated with github actions and see theses tests pass before merging

**Comment by chrissharp80** [comment]
@mrshah-at-ibm this PR would not work with Github Actions from a fork because the env variables are not exposed to a fork, even if it has the same ones set. As a result, I created a branch from the main repo and pushed my code there. I would like to have this PR closed.

File: integration_test/integration_functions_lib.go [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+	tlsCert, err := base64.StdEncoding.DecodeString(ec)
+	if err != nil {
+		Logger.Println(LogPrefix+"error copying the cert", err)
+		return nil, err
+	}
+	return tlsCert, nil
+}
+
+func WriteFileToLocalDirectory(filename string, tlsCert []byte) error {
+	// convert the tls cert from the newly created CA into a format that can be used to create a PEM file
```
Comment (mrshah-at-ibm) [inline integration_test/integration_functions_lib.go:86]: The comment for the function should go above the Function definition.

File: integration_test/integration_functions_lib.go [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+	}
+	if err := f.Sync(); err != nil {
+		Logger.Println(LogPrefix+"**ERROR** - problem during file sync: ", err)
+		return err
+	}
+	return nil
+}
+
+func CreateClient(tlsCertFilePath, apiURL, name string) *lib.Client {
+	// create a client config and enable it with TLS (using the tls cert from the CA's MSP)
```
Comment (mrshah-at-ibm) [inline integration_test/integration_functions_lib.go:111]: comment before function definition

File: integration_test/integration_functions_lib.go [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+	client := &lib.Client{
+		HomeDir: ".",
+		Config:  cfg,
+	}
+	Logger.Println("**SUCCESS** - client created")
+	return client
+}
+
+func EnrollCA(client *lib.Client, name, secret string) (*lib.EnrollmentResponse, error) {
+	// use the client to enroll the CA
```
Comment (mrshah-at-ibm) [inline integration_test/integration_functions_lib.go:132]: comment before function definition

File: integration_test/integration_functions_lib.go (hunk 4 above)
Comment (mrshah-at-ibm) [reply integration_test/integration_functions_lib.go:291]: cool. I think Poll or ticker would be a better implementation. But that should not block this.

**Review by mrshah-at-ibm — APPROVED** [review]

**Comment by dshuffma-ibm** [comment]
closing, dup of https://github.com/IBM-Blockchain/ibp-go-sdk/pull/12

**Pull request closed without merging**